### PR TITLE
feat(Marketplace): add looksrare

### LINKED ISSUE #5491: [Bounty] Integrate LooksRare, an NFT Marketplace into Mask Plugin
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [x] Intermediate
  - [ ] Advanced
- Category
  - [ ] Mask Core Feature
  - [x] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [ ] Other
    - [ ] /_List it here_/
- Time Commitment
  - [x] 2 weeks
- Bounty Size
  - [x] 150 $Mask or 2000 USD

## Part 1: Problem Description
**Background**
LooksRare is the community-first NFT marketplace on Ethereum that actively rewards traders, collectors and creators for participating

**Solution**
Create a plugin that integrates LooksRare. The basic product logic could resemble our Opensea and Rarible integration. The UI is in a card style with different tabs serving different purposes.

150004886-c7d03bf0-6099-4310-bb44-c9a8829f2f11

Default page could be the item overview
A second tab for detailed info
Other tabs for offerings, historical prices etc.

Please talk to us before committing to an integrated design. You could stick with the Opensea design.

Anylink from LooksRare on Twitter could trigger the plugin ( could have some slightly different treatment). The tab order can be changed, corresponding landing and ending page could be added. Share button for such a social nft platform is a must

Suggested Verbiage
"I just purchased xxx for yyy dollar, and I feel like I'm on top of the world" could be added to help spread the word

## Part 2: Resources

- [Bounty Guide](https://github.com/DimensionDev/Maskbook/tree/master/docs/bounty-guide.md)
- https://looksrare.org/

## Part 3: Contact and Support
@yisiliu @tzhan28 @LindaMask for ideas
@Jack-Works for implementations
Mask Network discord server: https://discord.gg/4Xccj7a75Q,
LooksRare discord server: https://discord.com/invite/looksrare

Please speak up in the Mask discord's bounty section or just discord dm @yisiliu or @taylor zhang or @lin_dao #5491 so that we can find you on Discord. You can also reach out to lin_DAO on telegram and she will create a dedicated group for faster communication

Submission Requirements
Please include demo to show the integration is working.

Judging Criteria
Have a working demo.
Clean code & tech design
Clear UI, easy to use and generally good user experience

### PR DESCRIPTION
**Pull request by developerfred**
## Description

Closes #5491

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/EVM/services/Asset/index.ts [hunk 1]
```diff
@@ -94,6 +98,8 @@ export async function getListings(options: ListOption) {
             return Rarible.getListings(address, tokenId, { chainId })
         case NonFungibleAssetProvider.ZORA:
             return Zora.getListings(address, tokenId)
+        case NonFungibleAssetProvider.LOOKSRARE:
+            return Zora.getListings(address, tokenId)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/EVM/services/Asset/index.ts:102]: ?

File: packages/web3-providers/package.json [hunk 2]
```diff
@@ -17,6 +18,7 @@
     "@masknet/web3-shared-evm": "workspace:*",
     "bignumber.js": "9.0.2",
     "date-fns": "^2.28.0",
+    "ethers": "^5.6.4",
```
Comment (guanbinrui) [inline packages/web3-providers/package.json:21]: It was not used.

File: packages/web3-providers/src/looksrare/index.ts [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        approved_on_chain: order.approved_on_chain,
+        listing_time: order.listing_time,
+        side: order.side,
+        quantity: order.quantity,
+        expiration_time: order.expiration_time,
+        order_hash: order.order_hash,
+    }
+}
+
+export class LooksRareAPI {
```
Comment (guanbinrui) [inline packages/web3-providers/src/looksrare/index.ts:238]:
```suggestion
export class LooksRareAPI implements NonFungibleTokenAPI.Provider {
```

File: packages/mask/src/plugins/EVM/services/Asset/index.ts (hunk 1 above)
Comment (developerfred) [reply packages/mask/src/plugins/EVM/services/Asset/index.ts:102]: I will change it when I implement the Looksrare getListings

**Pull request closed without merging**